### PR TITLE
feat: fetch on-chain decoder from configuration that is deploy by code_hash

### PR DESCRIPTION
**Pull request by ashuralyk**
# Description
for ultimate on-chain principle, we also need to download decoder binary on-chain that even it's deployed by `code_hash`, to achieve this, we associate decoder's code hash with its deployment information, specificly the `tx_hash` and `out_index`

### CODE REVIEW

File: src/decoder.rs [hunk 1]
```diff
@@ -236,6 +257,31 @@ impl DOBDecoder {
...
             .into())
     }
+
+    // search on-chain decoder cell, directly by its tx_hash and out_index
+    fn fetch_decoder_binary_directly(
+        &self,
+        tx_hash: H256,
+        out_index: usize,
+    ) -> DecodeResult<Vec<u8>> {
+        let decoder_deployment_tx = self
```
Comment (EthanYuan) [inline src/decoder.rs:267]: It's better to use this rpc [get_live_cell](https://github.com/nervosnetwork/ckb/tree/develop/rpc#method-get_live_cell)

File: src/decoder.rs [hunk 2]
```diff
@@ -236,6 +257,31 @@ impl DOBDecoder {
...
+            .ckb_rpc
+            .get_transaction(tx_hash)
```
Comment (EthanYuan) [inline src/decoder.rs:269]: It's better to use this rpc [get_live_cell](https://github.com/nervosnetwork/ckb/tree/develop/rpc#method-get_live_cell)

Comment (ashuralyk) [reply src/decoder.rs:275]: I'll try it

**Comment by ashuralyk** [comment]
@EthanYuan it's ready to merge?

**Pull request merged**